timApp/document/translation/deepl.py

```diff
@@ -368,10 +351,29 @@ def translate(
                 else requests.adapters.DEFAULT_RETRIES
             ),
         )
-        for i in range(0, len(protected_texts), 50):
+
+        i = 0
+        while i < len(protected_texts):
+            req_chars = 0
+            num_params = 0
+            start = i
+
+            if not len(protected_texts[i]) < DEEPL_REQUEST_SIZE_LIMIT:
+                i += 1
+            else:
+                # In addition to the text parameters limit, requests also have a size limit in bytes
+                while (
+                    i < len(protected_texts)
+                    and num_params < DEEPL_REQUEST_MAX_TEXT_PARAMS
+                    and req_chars + len(protected_texts[i]) < DEEPL_REQUEST_SIZE_LIMIT
+                ):
+                    req_chars += len(protected_texts[i])
+                    num_params += 1
+                    i += 1
```

Comment:
Kääntäisin nämä if/else:n järjestystä toisinpäin , sillä ehto `len(protected_texts[i]) < DEEPL_REQUEST_SIZE_LIMIT` on se "pääehto" ja vastapuoli on vain yhden rivin pituinen

eli

```py
if len(protected_texts[i]) < DEEPL_REQUEST_SIZE_LIMIT:
   # toinen while, joka etsii loppupään
else:
  i += 1
```

lisäksi tämä poistaa sen harhaanjohtavan `not a < b`.

Reply:
Joo, totta. Tämä korjattu.